model-fcrepo/src/test/java/edu/unc/lib/boxc/model/fcrepo/objects/FileObjectTest.java

```diff
@@ -0,0 +1,184 @@
+    @Captor
+    private ArgumentCaptor<Model> modelCaptor;
+    @Mock
+    private URI storageUri;
+
+    @BeforeEach
+    public void init() {
+        closeable = openMocks(this);
+        pid = pidMinter.mintContentPid();
+        fileObject = new FileObjectImpl(pid, driver, repoObjFactory);
+        model = ModelFactory.createDefaultModel();
+        resc = model.getResource(pid.getRepositoryPath());
+        resc.addProperty(RDF.type, PcdmModels.Object);
+        resc.addProperty(RDF.type, Cdr.FileObject);
+
+        types = List.of(PcdmModels.Object.getURI(), Cdr.FileObject.getURI());
+
+        when(driver.loadTypes(eq(fileObject))).thenAnswer(new Answer<RepositoryObjectDriver>() {
+            @Override
+            public RepositoryObjectDriver answer(InvocationOnMock invocation) throws Throwable {
+                fileObject.setTypes(types);
+                return driver;
+            }
+        });
+
+        when(driver.loadModel(eq(fileObject), anyBoolean())).thenAnswer(new Answer<RepositoryObjectDriver>() {
+            @Override
+            public RepositoryObjectDriver answer(InvocationOnMock invocation) throws Throwable {
+                fileObject.storeModel(model);
+                return driver;
+            }
+        });
+    }
+
+    @AfterEach
+    void closeService() throws Exception {
+        closeable.close();
+    }
+
+    @Test
+    public void isValidTypeTest() {
+        assertEquals(fileObject, fileObject.validateType());
+    }
+
+    @Test
+    public void invalidTypeTest() {
+        Assertions.assertThrows(ObjectTypeMismatchException.class, () -> {
+            types = Arrays.asList(PcdmModels.Object.getURI(), Cdr.Folder.getURI());
+            fileObject.validateType();
+        });
+    }
+
+    @Test
+    public void addOriginalFileTest() {
+        fileObject.addOriginalFile(storageUri, "file.txt", MediaType.TEXT_PLAIN_VALUE, null, null);
+        var originalFilePid = getOriginalFilePid(pid);
+
+        verify(repoObjFactory).createOrUpdateBinary(eq(originalFilePid), eq(storageUri), eq("file.txt"),
+                eq(MediaType.TEXT_PLAIN_VALUE), any(), isNull(), modelCaptor.capture());
+
+        assertTrue(modelCaptor.getValue().getResource(originalFilePid.getRepositoryPath())
+                .hasProperty(RDF.type, PcdmUse.OriginalFile));
+    }
+
+    @Test
+    public void replaceOriginalFileTest() {
+        fileObject.replaceOriginalFile(storageUri, "file.txt", MediaType.TEXT_PLAIN_VALUE, null, null);
+        var originalFilePid = getOriginalFilePid(pid);
+
+        verify(repoObjFactory).createOrUpdateBinary(eq(originalFilePid), eq(storageUri), eq("file.txt"),
+                eq(MediaType.TEXT_PLAIN_VALUE), any(), isNull(), modelCaptor.capture());
+
+        assertNull(modelCaptor.getValue());
+    }
+
+    @Test
+    public void addBinaryTest() {
+        var binPid = getTechnicalMetadataPid(pid);
+        fileObject.addBinary(binPid, storageUri, "file.txt",
```

Comment:
It'd be good to verify that addBinary returns the binary it created, at least for this baseline addBinary test